### DISCUSSION TITLE
How to retrieve sort type from the expressions returned by GetOrderingExpressions<T>()?

### DISCUSSION THREAD

**Discussion by sivaji55** [alirezanet/Gridify#156] [General]
How to retrieve the sort type from expressions returned by GetOrderingExpressions<T>()?, I am trying use which takes the gridify query and parses the order by expression and will create new order by string to consumer another api which supports gridify, 
Note: field names are different in other api that supports gridify.

**Comment by alirezanet** [comment]
Hi @sivaji55, 
I'm not sure if I understand correctly what you're looking for but if you need the target property type, you can use something like this:
```csharp
var orderingExpression = new GridifyQuery() {OrderBy = "NormalDateTime"}.GetOrderingExpressions<TestModel>();
var unaryExpression = (UnaryExpression) orderingExpression.First().Body;
var targetPropertyType = unaryExpression.Operand.Type;


class TestModel
{
	public DateTime? NormalDateTime { get; set; }
	public DateTimeOffset? DateTimeOffset { get; set; }
}
```

**Reply by sivaji55** [reply]
Hi @alirezanet, I am looking for Sort Direction (Ascending or Descending) in the expression returned by GetOrderingExpressions<T>(); method

**Reply by alirezanet** [reply]
I don't think getting the sort type from the ordering expression is possible since it is just a property selector that you can use later in an OrderBy or OrderByDecending methods. feel free to add a feature request if you need this information.

**Reply by moritzrinow** [reply]
@alirezanet I stumbled upon this in situations where I would like to extract the target property type and sort mechanism from a raw order-by string. After looking at the internal logic of `IGridifyOrdering.IsValid()` it seems to me that all this information would be easily exposed to consumers via the already existing `ParseOrderings` method returning a list of `ParsedOrdering`. Unfortunately this functionaly is marked as 'internal'. I would assume this functionality to be pretty stable, so why not just make it public? Imo this is a must-have when trying to implement special pagination mechanisms outside of Gridify. In this case, I would like to make use of a pagination mechanism using page-tokens, which are based on timestamps and not just simple skip/take. For that I need to know what kind of ordering is being applied by the user.

**Comment by georgopoulosgianniss** [comment]
Has anyone found a workaround to obtain the sort direction?

**Reply by alirezanet** [reply]
Hi @georgopoulosgianniss,
you can try something like:
```csharp
var parsedOrdering = SyntaxTree.ParseOrderings("name asc");
var isAscending = parsedOrdering.First().IsAscending;
```
